crates/sui-source-validation/src/lib.rs

```diff
@@ -0,0 +1,238 @@
+                }
+
+                let mod_str = module_symbol.to_string();
+                let on_chain_bytes = match on_chain_package.module_map.get(&mod_str) {
+                    Some(oc_bytes) => oc_bytes,
+                    None => {
+                        return Err(DependencyVerificationError::OnChainDependencyNotFound(
+                            pkg_symbol,
+                            Some(module_symbol),
+                        ))
+                    }
+                };
+
+                // compare local bytecode to on-chain bytecode to ensure integrity of our dependencies
+                if &local_bytes != on_chain_bytes {
+                    return Err(DependencyVerificationError::ModuleBytecodeMismatch(
+                        pkg_symbol.to_string(),
+                        module_symbol.to_string(),
+                        addr,
+                    ));
+                }
+
+                on_chain_module_count += 1;
+
+                if self.verbose {
+                    println!(
+                        "{}::{} - {} bytes, code matches",
+                        pkg_symbol,
+                        module_symbol,
+                        on_chain_bytes.len()
+                    );
+                }
+            }
+
+            verified_dependencies.insert(
+                addr,
+                Dependency {
+                    symbol: pkg_symbol.to_string(),
+                    module_bytes: on_chain_package.module_map.clone(),
+                },
+            );
+        }
+
+        // total number of modules in packages must match, in addition to each individual module matching
+        let len = compiled_package.deps_compiled_units.len();
+        // only need to check for less than, because if local modules are missing on-chain we've
+        // already errored out with a OnChainDependencyNotFound error
+        if len < on_chain_module_count {
+            let missing_modules = get_missing_modules(&compiled_package, &verified_dependencies);
+            return Err(DependencyVerificationError::ModuleCountMismatch(
+                len,
+                on_chain_module_count,
+                missing_modules,
+            ));
+        }
+
+        Ok(())
+    }
+
+    async fn pkg_for_address(
+        &mut self,
+        addr: &AccountAddress,
+    ) -> Result<SuiRawMovePackage, DependencyVerificationError> {
+        // Move packages are specified with an AccountAddress, but are
+        // fetched from a sui network via sui_getObject, which takes an object ID
+        let obj_id = ObjectID::from(*addr);
+
+        // fetch the Sui object at the address specified for the package in the local resolution table
+        // if future packages with a large set of dependency packages prove too slow to verify,
+        // batched object fetching should be added to the ReadApi & used here
+        let obj_read = match self.rpc_client.get_object(obj_id).await {
+            Ok(raw) => raw,
```

Comment:
More spots for `let-else`! Hurray! (Everyone is as excited about `let-else` right?)

Reply:
Alternatively you can do `.map_err(DependencyVerificationError::DependencyObjectReadFailure)` which might be a bit more idiomatic Rust